src/app/admin/packingSlotsTable/PackingSlotActions.ts

```diff
@@ -0,0 +1,83 @@
+import supabase from "@/supabaseClient";
+import { DatabaseError } from "@/app/errorClasses";
+import { GridRowId } from "@mui/x-data-grid";
+import { PackingSlotRow } from "@/app/admin/packingSlotsTable/PackingSlotsTable";
+import { Tables } from "@/databaseTypesFile";
+import { logError } from "@/logger/logger";
+
+type DbPackingSlot = Tables<"packing_slots">;
+type NewDbPackingSlot = Omit<DbPackingSlot, "primary_key">;
+
+export const fetchPackingSlots = async (): Promise<PackingSlotRow[]> => {
+    const { data, error } = await supabase.from("packing_slots").select().order("order");
+    if (error) {
+        throw new DatabaseError("fetch", "packing slots");
+    }
+
+    return data.map(
+        (row): PackingSlotRow => ({
+            name: row.name,
+            id: row.primary_key,
+            isShown: row.is_shown,
+            order: row.order,
+            isNew: false,
+        })
+    );
+};
+
+const formatExistingRowToDBPackingSlot = (row: PackingSlotRow): DbPackingSlot => {
+    return {
+        primary_key: row.id,
+        name: row.name,
+        is_shown: row.isShown,
+        order: row.order,
+    };
+};
+
+const formatNewRowToDBPackingSlot = (newRow: PackingSlotRow): NewDbPackingSlot => {
+    return {
+        name: newRow.name,
+        is_shown: newRow.isShown,
+        order: newRow.order,
+    };
+};
+
+export const dbPackingSlotToInsert = async (newRow: PackingSlotRow): Promise<void> => {
+    const data = formatNewRowToDBPackingSlot(newRow);
+    const { error } = await supabase.from("packing_slots").insert(data);
+
+    if (error) {
+        throw new DatabaseError("insert", "packing slots");
+    }
+};
+
+export const dbPackingSlotToUpdate = async (row: PackingSlotRow): Promise<void> => {
+    const processedData = formatExistingRowToDBPackingSlot(row);
+    const { error } = await supabase
+        .from("packing_slots")
+        .update(processedData)
+        .eq("primary_key", processedData.primary_key);
+
+    if (error) {
+        throw new DatabaseError("update", "packing slots");
+    }
+};
+
+export const dbPackingSlotToDelete = async (id: GridRowId): Promise<void> => {
```

Comment:
We don't actually ever want to delete a packing slot from the database do we??

Reply:
What if one admin keeps adding new slots and there's no way of managing a huge list of slots?

Reply:
I do see the point - however I think they'd be more concerned about accidentally deleting a packing slot, than having too many. I think hiding the packing slot is good enough at least for now.